injector/src/main/java/edu/ucr/cs/riple/injector/offsets/OffsetChange.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022 Nima Karimipour
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.injector.offsets;
+
+import java.util.List;
+
+/**
+ * Offset change information, stores the number of character added / removed at a specific position.
+ */
+public class OffsetChange {
+
+  /** Index of addition / deletion. */
+  public final int position;
+  /** Number of characters added / removed. */
+  public final int dist;
+
+  public OffsetChange(int position, int dist) {
+    this.position = position;
+    this.dist = dist;
+  }
+
+  /**
+   * Computes the original offset of the given offset according to existing offset changes.
+   *
+   * @param offset Given offset change.
+   * @param existingOffsetChanges Existing offsets.
+   * @return Original offset.
+   */
+  public static int getOriginalOffset(int offset, List<OffsetChange> existingOffsetChanges) {
+    if (existingOffsetChanges == null) {
+      return offset;
+    }
+    for (OffsetChange current : existingOffsetChanges) {
+      if (offset > current.position) {
+        offset -= current.dist;
+      }
+      if (offset < current.position) {
+        break;
```

Comment:
The way I thought about it is that `Injector` is only responsible for informing the user about the file changes and it should be responsibility of user to call the APIs with correct arguments. In `annotator-core` module, we store the recent updates reported by the injector and always keep them sorted. Please see [here](https://github.com/ucr-riple/NullAwayAnnotator/blob/61ef4ce13e4cbde601cda4851a1336d444ef46dc/annotator-core/src/main/java/edu/ucr/cs/riple/core/Config.java#L767).

I added a comment to emphasize that the input must be sorted. 375ebadc6d8343d49b3de3f07eda50e221d3c2ad

Reply:
I am a bit concerned about how easy it's going to be to maintain this cross-*module* invariant without any sort of datastructure guarantees or checking...

Reply:
Agreed with @lazaroclapp.  Maybe we can just take a `SortedSet` as an argument, with `TreeSet` as the implementation?  We can sleep better knowing some change elsewhere won't break this code in a subtle way.  And I really don't think there will be much performance impact.

Reply:
@msridhar @lazaroclapp We can't use `Sets` here as we might have duplicates in this collection. We might need to add two annotations on the same element with identical character length such as:
```java
@Annot1 @Annot2 Object foo;
```

This is not the case for now and we only care about `@Nullable` annotations at this moment, Please let me know if you think we can move forward with sets now and update the structure later if needed and I will update this.

The other alternative is to move the logic of updating offsets as is from `injector` to `annotator-core` completely so it won't be a cross-module implementation. Please let me know which one you think works best.

Reply:
I personally would prefer sticking with `SortedSet` and for the case you mention, adding the injected / deleted `String` as another field in `OffsetChange` to disambiguate.  Or there could be fields for the position and the text, and then num characters could be computed and cached if necessary.  I like `SortedSet` since then the type system helps enforce and document the property we need.

Reply:
ef3569b910409e50d4ccdd068cac9125b4b00b32